### ISSUE TITLE
Does `getServerSession` work with Route Handlers?

### ISSUE THREAD

**Issue by rijk** [nextauthjs/next-auth#7355]
### Question 💬

Implementing Next's new [Route Handlers](https://beta.nextjs.org/docs/routing/route-handlers), I tried to do the following:

```ts
export async function GET(req: NextRequest) {
  const session = await getServerSession(authOptions)
  console.log('session', session?.user.email)
}
```

This works for me locally, but not when deployed on Vercel (it's always giving me `session undefined`).

Is this supported? What should I do to make it work?

### How to reproduce ☕️

See the snippet above — can provide a repro but as said it only happens when deployed on Vercel I think.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by mike14747** [comment]
I've noticed this too using getServerSession() in server components while using a route handler for next-auth in the appDir. It works locally, but not when deployed to Vercel.

Funny thing is that useSession() would still work in client components... just no getServerSession() in server components.

So, for now, I've switched back to the /pages/api/auth directory for next-auth.

**Comment by mike14747** [comment]
Thanks for the response. I just switched one of my apps back to next-auth in the appDir using a route handler. Before I did that, I updated next from 13.3.0 to v13.3.1.

To my surprise, it's now working on vercel. Was there something in that update that fixes the issue I was having? Not sure.

I'm going to check out a couple other apps and see if they're able to now use getServerSession() with a next-auth route handler. I'll post here again afterward.
